src/physics/nonlinear_solid.cpp

```diff
@@ -199,6 +197,10 @@ void NonlinearSolid::completeSetup()
     // ordinary differential equation. Here, we define the residual function in
     // terms of an acceleration.
     residual.function = [=](const mfem::Vector& d2u_dt2, mfem::Vector& res) mutable {
+
+      // TODO: we should avoid re-assembling this when not required
+      C_mat.reset(C->ParallelAssemble());
+
       res = (*M_mat) * d2u_dt2 + (*C_mat) * (du_dt + c1 * d2u_dt2) + (*H) * (x + u + c0 * d2u_dt2);
```

Comment:
This is a "nonlinear solid" but it's really being formulated as a linear solid, with C and H.  How does that work?

Reply:
Maybe that's the function pointer "H".  But you would really want it to be a function of both u and v, to accomodate non-linear viscoelastic/viscoplastic effects

Reply:
Most of these changes to the NonlinearSolid class are just reimplementing the ODE with less boilerplate code, rather than making it support more material models. You're right that in the longer term we need to provide a way to pass both displacements and velocities to target more constitutive models, but Serac isn't quite there yet. However, Julian has been doing some important work recently that would give us a more elegant way to define materials involving multiple finite element fields (e.g. pressure/velocity, displacement/velocity, efield/bfield, etc), so stay tuned!